src/prpy/planning/snap.py

```diff
@@ -106,7 +110,11 @@ def _Snap(self, robot, goal, **kw_args):
         # Make the trajectory end at the goal configuration, as
         # long as it is not in collision and is not identical to
         # the start configuration.
-        CheckJointLimits(robot, goal, deterministic=True)
+        try:
+            CheckJointLimits(robot, goal, deterministic=True)
+        except SoftJointLimitError:
+            pass
```

Comment:
Well doesn't `raise` mean throwing the error? And `CheckJointLimits` does raise an error, we were just ignoring it. This may change with the new API suggested by Chris though. I've reverted `CheckJointLimits` to just return a `SoftJointLimitError`. But it seems that this is no longer necessary anyway since the `HardwareJointLimitValidator` will take care of that case?